### PR TITLE
duplicacy: init at 2.1.2

### PR DESCRIPTION
**Pull request by ffinkdevs**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/backup/duplicacy/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "duplicacy-${version}";
```
Comment (Mic92) [inline pkgs/tools/backup/duplicacy/default.nix]:
```suggestion
  pname = "duplicacy";
```

File: pkgs/tools/backup/duplicacy/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "0v3rk4da4b6dhqq8zsr4z27wd8p7crxapkn265kwpsaa99xszzbv";
+  };
+  goDeps = ./deps.nix;
+  buildPhase = ''
+    cd go/src/${goPackagePath}
+    go build duplicacy/duplicacy_main.go
+  '';
+
+  installPhase = ''
+    mkdir -p $bin/bin
```
Comment (Mic92) [inline pkgs/tools/backup/duplicacy/default.nix]:
```suggestion
```

File: pkgs/tools/backup/duplicacy/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    cp duplicacy_main $bin/bin/duplicacy
```
Comment (Mic92) [inline pkgs/tools/backup/duplicacy/default.nix]:
```suggestion
    install -D duplicacy_main $bin/bin/duplicacy
```

**Review by Mic92 — CHANGES_REQUESTED** [review]
Looks good to. I only left some nitpicks. Could not test the package because of my flacky connectivity.

**Comment by Mic92** [comment]
@GrahamcOfBorg build duplicacy

**Comment by ffinkdevs** [comment]
@Mic92  you sure its pname when using go packages?, i pretty much just copied restic's derivation and changed it to duplicacy needs

**Comment by Mic92** [comment]
Indeed this was a limitation of `buildGoPackage` to not exposing this feature of stdenv.

**Comment by Mic92** [comment]
@GrahamcOfBorg eval

**Comment by Ralith** [comment]
What's the status of this?

**Comment by Mic92** [comment]
Sorry. I forgot about this one.

**Pull request merged**